src/ed_object_models/spawn_sdf_in_gazebo.py

```diff
@@ -0,0 +1,83 @@
+#! /usr/bin/python
+
+import rospy
+import yaml
+import os
+import glob
+
+from rospkg import RosPack
+from gazebo_msgs.srv import SpawnModel
+from geometry_msgs.msg import Pose
+
+
+def from_yaml(yaml_file):
+    ''' Spawns a list of sdf files from a yaml file into Gazebo.
+    :param yaml_file:
+    :return:
+    '''
+
+    # Initialize ROS node
+    rospy.init_node('gazebo_simulator_object_spawner')
+
+    # Wait until gazebo is ready to spawn sdf models
+    rospy.wait_for_service('gazebo/spawn_sdf_model')
+    spawn_model_prox = rospy.ServiceProxy('gazebo/spawn_sdf_model', SpawnModel)
+
+    # Load yaml with list of objects that need to be loaded
+    yaml_file_path = None
+    if os.path.isfile(yaml_file):
+        yaml_file_path = yaml_file
+    else:
+        package_path = RosPack().get_path('fast_simulator_data')
+        if os.path.isfile(package_path + yaml_file):
+            yaml_file_path = package_path + yaml_file
+
+    if yaml_file_path is None:
+        print('Warning: Could not find yaml file.')
+        return
+
+    with open(yaml_file_path, 'r') as f:
+        data = yaml.safe_load(f)
+
+    # Get paths in $GAZEBO_MODEL_PATH
+    model_paths = os.environ['GAZEBO_MODEL_PATH'].split(os.pathsep)
+
+    # Iterate over objects and spawn
+    for sod in data:
+        # Define object pose
+        object_pose = Pose()
+        object_pose.position.x = sod['x']
+        object_pose.position.y = sod['y']
+        object_pose.position.z = sod['z']
+
+        # Search for model folder in $GAZEBO_MODEL_PATH
+        model_path = None
+        for path in model_paths:
+            if os.path.isdir(path + '/' + sod['type']):
+                model_path = path + '/' + sod['type']
```

Comment:
There is `os.path.join()` or the `os.sep` character for creating a file system path.